### ISSUE TITLE
<Link /> is an object instead of a React Component since 4.19 making tests fail

### ISSUE THREAD

**Issue by openscript** [gatsbyjs/gatsby#36298]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

After I upgraded from Gatsby `4.18.2` to `4.19.2` and with this `gatsby-link` to `4.19.1` my tests using the `<Link />` component start to fail:

```
    console.error
      Warning: React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: object.
      
      Check the render method of `LocalizedLink`.
          at to (/workspaces/gatsby-plugin-i18n-l10n/src/components/LocalizedLink.tsx:17:3)
          at IntlProvider (/workspaces/gatsby-plugin-i18n-l10n/node_modules/react-intl/src/components/provider.js:84:47)

      29 |   const prefixedPath = intl.defaultLocale === intl.locale ? localizedPath : trimRightSlash(nodePath.join('/', prefix, localizedPath));
      30 |   return (
    > 31 |     <Link
         |     ^
      32 |       to={prefixedPath}
      33 |       className={className}
      34 |       activeClassName={activeClassName}
```

### Reproduction Link

https://github.com/openscript-ch/gatsby-plugin-i18n-l10n

### Steps to Reproduce

1. Set up Jest testing with Typescript according to the guides
2. Use Link component

### Expected Result

The test renderer can render `<Link />` as a React component and doesn't throw the error mentioned above.

### Actual Result

Test renderer cannot render `<Link />` as it seems to be an object now.

### Environment

```shell
System:
    OS: Linux 5.18 Debian GNU/Linux 11 (bullseye) 11 (bullseye)
    CPU: (8) x64 Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
    Shell: 5.1.4 - /bin/bash
  Binaries:
    Node: 18.0.0 - /usr/local/bin/node
    Yarn: 1.22.18 - /usr/bin/yarn
    npm: 8.6.0 - /usr/local/bin/npm
  Languages:
    Python: 3.9.2 - /usr/bin/python
  npmPackages:
    gatsby: ^4.19.2 => 4.19.2
    gatsby-source-filesystem: ^4.19.0 => 4.19.0
```

**Comment by openscript** [comment]
This also happens with Gatsby 4.20.

**Comment by pieh** [comment]
Can confirm / reproduce. Opened https://github.com/gatsbyjs/gatsby/pull/36311 with potential fix for it

**Comment by pieh** [comment]
I'm not sure if the change in PR is correct (hence the draft state of it). In the meantime, as a workaround, you can add:

```
'^gatsby-link$': `<rootDir>/node_modules/gatsby-link/dist/index.modern.mjs`,
```

to your `moduleNameMapper` map in `jest.config.js` - https://github.com/openscript-ch/gatsby-plugin-i18n-l10n/blob/a4b691f9e6858db6b49d14f81f593ce46b7edba1/jest.config.js#L5-L8

so that it uses ES variant instead of CJS (that has problems I outlined in my PR description)

**Comment by pieh** [comment]
Fix was published in `gatsby@next` ( concretely `gatsby@4.21.0-next.2` ). I won't be backporting it to stable for now to monitor for sometime if it didn't introduce different kind of quirks/regressions
